### ISSUE TITLE
no-return-assign rule has other default option?

### ISSUE THREAD

**Issue by SerhiiBilyk** [eslint/eslint#11049]
Environment:

eslint: 5.8.0
node : 10.10.0
npm: 6.4.0

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  "extends": "airbnb",
  "settings": {
    "react": {
      "version": "16.6.0"
    }
  },
  "parser": "babel-eslint",
  "plugins": [
    "jsdoc",
  ],
  "parserOptions": {
    "sourceType": "module"
  },
  "rules": {
    "react/destructuring-assignment": [true, 'always'],
    "react/no-access-state-in-setstate": ["error"],
    "react/no-children-prop": ["error"],
    "react/no-danger": ["error"],
    "react/no-danger-with-children": ["error"],
    "react/no-redundant-should-component-update": ["error"],
    "react/boolean-prop-naming": ["error", {
      "rule": "^is[A-Z]([A-Za-z0-9]?)+"
    }],
    "no-restricted-syntax": ["warn", {
      selector: "UnaryExpression[operator='delete'] > MemberExpression",
      message: "Always check if [delete] returns [true]"
    }],
    "require-jsdoc": ["error", {
      "require": {
        "FunctionDeclaration": true,
        "MethodDefinition": false,
        "ClassDeclaration": false,
        "ArrowFunctionExpression": false,
        "FunctionExpression": false
      }
    }],  
    "eqeqeq": 2,
    "no-dupe-keys": 1,
    "no-caller": 2,
    "no-console": 1,
    "no-else-return": 2,
    "no-empty-function": 2,
    "no-extend-native": 2,
    "no-loop-func": 2,
    "no-new-func": 2,
    "no-new-wrappers": 2,
    "no-param-reassign": 2,
    "no-return-assign": ["error"],
    "no-useless-return": 2,
    "yoda": 2,
    "no-nested-ternary": 2,
    "prefer-template": 2,
    "prefer-destructuring": 1,
    "no-delete-var": 2,
    "no-redeclare": 2,
    "default-case": 2,
    "prefer-arrow-callback": 1,
    "no-undef-init": 2,
    "no-alert": 2,
    "no-implied-eval": 2,
    "no-use-before-define": 2,
    "linebreak-style": ["error", "windows"],
    "no-unused-vars": ["error", {
      "vars": "all",
      "args": "after-used",
      "ignoreRestSiblings": false,
      "varsIgnorePattern": "(_jsDoc|React)$"
    }],
    "jsdoc/check-param-names": 1,
    "jsdoc/check-tag-names": 1,
    "jsdoc/check-types": 1,
    "jsdoc/require-returns-type": 1,
    "jsdoc/valid-types": 1
  },
  "globals": {
    angular: false,
    jQuery: false,
    $: false,
    moment: false,
    app: false,
    __NODE_ENV__: false,
    React: false,
    /**
     * Jest test framework
     */
    describe: false,
    beforeAll: false,
    afterAll: false,
    beforeEach: false,
    it: false,
    jest: false,
    expect: false,
  },
  "env": {
    "browser": true,
    "node": true,
    "es6": true
  }
}
```

</details>

I' using VSCode IDE.
Due to documentation,  "no-return-assign" has default option "except-parens".
**What did you expect to happen?**
But my code 
```js
this.props.handleInput('', () => (this.input.value = ''));
```
is highlighted with message : "[eslint] Arrow function should not return assignment"

With option : "no-return-assign": ["error","except-parens"] it works as expected.

**Are you willing to submit a pull request to fix this bug?**
I can submit a PR.
Like I understood, by default this rule uses "always" option. Please provide some information for me where I can find in source code default option merging. Thank you

**Comment by platinumazure** [comment]
Hi @SerhiiBilyk, thanks for the issue.

The `eslint-config-airbnb` project uses the "always" option in their style guide. I *believe* that ESLint core, when it merges extending configs, will not override options set in an extended config if the extending config only has the rule severity. (I wish this would only happen when the string severity is used, but I think it might happen with 1-length array severities are used too.)

In order to confirm this, could you run ESLint with the `--print-config` option? This will show the fully resolved config and you'll be able to see if `no-return-assign` still has the "always" option specified. If it does, you can fix this by explicitly setting "except-parens" in your config. If it doesn't, then we might have a bug in the rule.

**Comment by SerhiiBilyk** [comment]
@platinumazure thank you sir for your fast answer.
I think you are right. Tomorrow I will check this.
Have a nice day

**Comment by SerhiiBilyk** [comment]
@platinumazure please close the issue. It is works es axpected

**Comment by platinumazure** [comment]
Thanks @SerhiiBilyk, glad we could figure it out.
